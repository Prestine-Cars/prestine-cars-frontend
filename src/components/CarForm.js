import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate, useParams } from 'react-router-dom';
import newCar from '../redux/actions/car';

const CarForm = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { cityId } = useParams();

  const [car, setCar] = useState({
    model: '',
    photo: '',
    description: '',
    cost: '',
    city: cityId,
  });

  const handleChange = (e) => {
    setCar({
      ...car,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(newCar(car, navigate));
  };

  return (
    <div>
      <h1>Add Car for this city</h1>
      <form>
        <input
          type="text"
          id="model"
          name="model"
          placeholder="Model Car"
          value={car.model}
          onChange={handleChange}
          minLength="3"
          maxLength="50"
          required
        />

        <input
          type="url"
          id="photo"
          name="Photo of Car"
          placeholder="photo"
          value={car.photo}
          onChange={handleChange}
          required
        />

        <input
          type="text"
          id="description"
          name="description"
          placeholder="Description"
          value={car.description}
          onChange={handleChange}
        />

        <input
          type="number"
          id="cost"
          name="cost"
          placeholder="Cost"
          value={car.cost}
          onChange={handleChange}
          required
        />
        <button type="submit" onClick={handleSubmit}>Add car</button>
      </form>
    </div>
  );
};

export default CarForm;
