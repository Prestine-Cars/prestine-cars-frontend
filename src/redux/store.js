import { configureStore } from '@reduxjs/toolkit';
import userReducer from './reducers/user';
import cityReducer from './reducers/city';
import citiesReducer from './reducers/cities';
import carReducer from './reducers/car';

const store = configureStore({
  reducer: {
    user: userReducer,
    cities: citiesReducer,
    city: cityReducer,
    car: carReducer,
  },
});

export default store;
