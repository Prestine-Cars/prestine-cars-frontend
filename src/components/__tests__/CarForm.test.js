import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import CarForm from '../CarForm';

describe('Test presence of Car Form', () => {
  test('should render Car Form component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <CarForm />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
