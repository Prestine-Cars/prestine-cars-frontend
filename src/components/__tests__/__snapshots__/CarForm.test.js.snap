// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Car Form should render Car Form component 1`] = `
<div
  className="d-flex flex-column justify-content-center align-items-center vh-100"
>
  <h2>
    Add Car for this city
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-outline form-white mb-4"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        <input
          className="form-control form-control-lg"
          id="model"
          maxLength="50"
          minLength="3"
          name="model"
          onChange={[Function]}
          placeholder="Model Car"
          required={true}
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="form-outline form-white mb-4"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        <input
          className="form-control form-control-lg"
          id="photo"
          name="photo"
          onChange={[Function]}
          placeholder="Photo of Car"
          required={true}
          type="url"
          value=""
        />
      </label>
    </div>
    <div
      className="form-outline form-white mb-4"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        <input
          className="form-control form-control-lg"
          id="description"
          name="description"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="form-outline form-white mb-4"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        <input
          className="form-control form-control-lg"
          id="cost"
          name="cost"
          onChange={[Function]}
          placeholder="Cost"
          required={true}
          type="number"
          value=""
        />
      </label>
    </div>
    <button
      className="bg-lime-500 text-white hover:bg-lime-400 px-6 py-2 rounded-full font-semibold text-decoration-none text-center"
      type="submit"
    >
      Add car
    </button>
  </form>
</div>
`;
